test/testcommon/src/main/java/io/pravega/test/common/TestUtils.java

```diff
@@ -48,4 +53,87 @@ public static int getAvailableListenPort() {
         throw new IllegalStateException(
                 String.format("Could not assign port in range %d - %d", BASE_PORT, MAX_PORT_COUNT + BASE_PORT));
     }
+
+    public static void testBlocking(RunnableWithException blockingFunction, Runnable unblocker) {
+        final AtomicReference<Exception> exception = new AtomicReference<>(null);
+        final Semaphore isBlocked = new Semaphore(0);
+        Thread t = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    blockingFunction.run();
+                } catch (Exception e) {
```

Comment:
`Throwable` since we may want to include `AssertionErrors` too.

Reply:
fixed